src/openforms/formio/tests/validation/test_selectboxes.py

```diff
@@ -0,0 +1,163 @@
+from django.test import SimpleTestCase
+
+from hypothesis import given, strategies as st
+
+from openforms.typing import JSONValue
+
+from ...typing import SelectBoxesComponent
+from .helpers import extract_error, validate_formio_data
+
+
+class SelectboxesValidationTests(SimpleTestCase):
+
+    def test_selectboxes_required_validation(self):
+        component: SelectBoxesComponent = {
+            "type": "selectboxes",
+            "key": "foo",
+            "label": "Test",
+            "validate": {"required": True},
+            "openForms": {"dataSrc": "manual"},  # type: ignore
+            "values": [
+                {"value": "a", "label": "A"},
+                {"value": "b", "label": "B"},
+                {"value": "c", "label": "C"},
+            ],
+        }
+
+        invalid_values = [
+            ({}, "required"),
+            ({"foo": None}, "null"),
+            # verified with https://formio.github.io/formio.js/app/builder.html -
+            # selectboxes being required means that at least one option must be checked
+            ({"foo": {"a": False, "b": False, "c": False}}, "required"),
+        ]
+
+        for data, error_code in invalid_values:
+            with self.subTest(data=data):
+                is_valid, errors = validate_formio_data(component, data)
+
+                self.assertFalse(is_valid)
+                self.assertIn(component["key"], errors)
+                error = extract_error(errors, component["key"])
+                self.assertEqual(error.code, error_code)
+
+    def test_optional_selectboxes(self):
+        # hypothetical input - it appears that the Formio.js SDK always provides the
+        # nested object, whether required or not.
```

Comment:
nested object = `"validate": {...}`?

Reply:
No, if you have a selectboxes component with values `a` and `b`, it will always send `[key]: {a: bool, b: bool}`. The object with keys `a` and `b` is the nested object that I refer too.

This is in contrast to radio which will send `[key]: 'a' | 'b'` or select multiple which sends `[key]: ['a', 'b']`